apps/plugin-browser/src/pages/popup/popup-contents/action-center/automated/select-scope/rows-by-location.tsx

```diff
@@ -0,0 +1,346 @@
+          return acc;
+        }
+
+        const newDomains = existingRule.restrictToDomains.filter(
+          (domain) => domain !== restrictToDomain,
+        );
+        acc[existingRule.entityTypeId] = {
+          ...existingRule,
+          restrictToDomains: newDomains,
+        };
+      } else {
+        acc[existingRule.entityTypeId] = existingRule;
+      }
+      return acc;
+    }, {});
+
+    setInferenceConfig({
+      ...inferenceConfig,
+      rules: Object.values(rulesByType),
+    });
+  };
+
+  return (
+    <TableRow>
+      <TableCell>
+        <SelectDomains
+          multiple={false}
+          options={domainOptions}
+          selectedDomains={restrictToDomain ? [restrictToDomain] : []}
+          setSelectedDomains={(newDomains) => {
+            updateDomain(newDomains[0]);
+          }}
+        />
+      </TableCell>
+      <TableCell>
+        <EntityTypeSelector
+          inputHeight="auto"
+          multiple
+          setTargetEntityTypeIds={(newTargetIds) => {
+            updateEntityTypeIds(newTargetIds);
+          }}
+          targetEntityTypeIds={entityTypeIds}
+        />
+      </TableCell>
+      <TableCell sx={{ width: 20, p: `4px !important` }}>
+        {entityTypeIds.length > 0 && (
+          <IconButton onClick={() => removeRule()} size="small">
+            <CloseIcon />
+          </IconButton>
+        )}
+      </TableCell>
+    </TableRow>
+  );
+};
+
+export const RowsByLocation = (props: CommonRowsProps) => {
+  const {
+    domainOptions,
+    draftRule,
+    setDraftRule,
+    inferenceConfig,
+    setInferenceConfig,
+  } = props;
+
+  const { rules } = inferenceConfig;
+
+  const rulesByLocation = useMemo<RuleByLocation[]>(() => {
+    const sortedRules: RuleByLocation[] = [];
+
+    for (const rule of rules) {
+      for (const domain of rule.restrictToDomains.length > 0
+        ? rule.restrictToDomains
+        : [""]) {
+        const existingRule = sortedRules.find(
+          (rle) => rle.restrictToDomain === domain,
+        );
+        if (existingRule) {
+          existingRule.entityTypeIds.push(rule.entityTypeId);
+          existingRule.entityTypeIds.sort();
+        } else {
+          sortedRules.push({
+            entityTypeIds: [rule.entityTypeId],
+            restrictToDomain: domain,
+          });
+        }
+      }
+    }
+
+    if (draftRule) {
+      sortedRules.push({
+        entityTypeIds: [],
+        restrictToDomain: draftRule.restrictToDomains[0] ?? "",
+      });
+    }
+
+    sortedRules.sort((a, b) => {
+      // Keep the draft at the bottom
+      if (a.entityTypeIds.length === 0) {
+        return 1;
+      }
+      if (b.entityTypeIds.length === 0) {
+        return -1;
+      }
+
+      return (a.restrictToDomain ?? "") > (b.restrictToDomain ?? "") ? 1 : -1;
+    });
+
+    return sortedRules;
+  }, [draftRule, rules]);
+
+  return (
+    <>
+      {rulesByLocation.map((rule) => (
```

Comment:
If we can, might be nice to sort by when a row was added to prevent re-orderings whilst making changes:

https://github.com/hashintel/hash/assets/42802102/c66480c8-8b61-4709-86f6-4ac493f5c3b0

Reply:
I agree this would be nice – I've added H-1786 to come back to it (either today if I get time, or in Jan if I don't)